src/dwarf/Gparser.c

```diff
@@ -574,6 +584,8 @@ get_rs_cache (unw_addr_space_t as, intrmask_t *saved_maskp)
   if ((atomic_read (&as->cache_generation) != atomic_read (&cache->generation))
        || !cache->hash)
     {
+      /* cache_size is only set in the global_cache, copy it over before flushing */
+      cache->log_size = as->global_cache.log_size;
```

Comment:
Hmm, setting the cache size may not work quite right with a tls cache - the GET_MEMORY calls will never get cleaned up ever, and will just leak if a thread is destroyed after setting a larger cache size?